algorithms/decide_gi.py

```diff
@@ -0,0 +1,60 @@
+from supporting_components.graph import Graph
+from supporting_components.graph import Vertex
+
+def is_balanced_or_bijected(G: 'Graph'):
+    """
+    :param G:
+    :return: bool is_balanced, bool is_bijected
+    This function returns if the graph is balanced and bijected.
+    Note: Only a balanced graph can be bijected.
+    """
+    is_balanced, colors = __is_balanced(G)
+    if is_balanced:
+        return is_balanced, __is_bijected(colors)
+
+    return False, False
+
+def __is_balanced(G: 'Graph'):
+    """
+    :param G:
+    :return: bool is_balanced, list g_self_colornums
+
+    The vertex coloring of the `self` and `other` graph are extracted from the disjoint union.
+    The extraction of `self` and `other` works via the attribute `graph_label` from Vertex.
+    If `graph_label` == True the vertex colornum is stored in the g_self_colornums list.
+    If `graph_label` == False the vertex colornum is stored in the g_other_colornums list.
+
+    To determine if the coloring is balanced, the lists g_self_colornums and g_other_colornums are compared for equality element-wise.
+    """
+    g_self_colornums  = []
+    g_other_colornums = []
+
+    for v in G.vertices:
+        if type(v.graph_label) == bool:
+            if v.graph_label == True:
+                g_self_colornums.append(v.colornum)
+            else:
+                g_other_colornums.append(v.colornum)
+        else:
+            print("Error: Vertex graph_label is not of type bool")
```

Comment:
Misschien dan ook even kijken of we dan het hele proces kunnen stoppen, anders gaat hij hierna gewoon door zonder deze vertex toe te voegen

Reply:
is een exception gooien een idee ?
raise ValueError('Error: Vertex with graph_label {graph_label} is not of type int"'.format(graph_label=repr('niet-een-int')))